GraphsV2/ChordGraph/index.js

```diff
@@ -0,0 +1,261 @@
+        disableRibbonHover,
+        strokeWidth,
+        resizeWithWindow,
+        blurOnHover,
+        ribbonOpacity,
+        ribbonBlurOpacity,
+        chordWeightColumn,
+        chordSourceColumn,
+        chordDestinationColumn,
+        colors,
+        additionalKeys,
+        bidirectionalTooltip,
+        id
+    } = properties
+
+    useEffect(() => {
+        setCustomTooltips(customTooltip(properties));
+        parseData(props.data);
+    }, [props.data, props.width, props.height]);
+
+    const parseData = (data) => {
+
+        const filterChordData = data.filter(d => d[chordSourceColumn] && d[chordDestinationColumn])
+
+        if (!isEqual(filterChordData, filterData)) {
+            setFilterData(filterChordData);
+        }
+    }
+
+    const clearHover = () => {
+        setStateHoverPersist(false);
+        setStateMouseOverGroup(null);
+        setStateMouseOverRibbon(null);
+    }
+
+    const setHoverPersist = (hoverPersist) => {
+        setStateHoverPersist(hoverPersist);
+    }
+
+    const setMouseOverGroup = (mouseOverGroup) => {
+        setStateMouseOverGroup(mouseOverGroup);
+    }
+
+    const weight = function (d) { return d[chordWeightColumn]; };
+    const source = function (d) { return d[chordSourceColumn]; };
+    const destination = function (d) { return d[chordDestinationColumn]; };
+    // Generates a matrix (2D array) from the given data, which is expected to
+    // have fields {origin, destination, count}. The matrix data structure is required
+    // for use with the D3 Chord layout.
+
+    const generateMatrix = (data) => {
+        if (isEmpty(data)) {
+            return []
+        }
+
+        var indices = {},
+            matrix = [],
+            names = [],
+            matrixData = [],
+            n = 0, i, j;
+
+        function recordIndex(name) {
+            if (!(name in indices)) {
+                indices[name] = n++;
+                names.push(name);
+            }
+        }
+
+        data.forEach(function (d) {
+            recordIndex(source(d));
+            recordIndex(destination(d));
+        });
+
+        for (i = 0; i < n; i++) {
+            matrix.push([]);
+            matrixData.push([]);
+            for (j = 0; j < n; j++) {
+                matrix[i].push(0);
+                matrixData[i].push([]);
+            }
+        }
+
+        data.forEach(function (d) {
+            i = indices[source(d)];
+            j = indices[destination(d)];
+
+            if (chordWeightColumn) {
+                matrix[j][i] += weight(d);
+            } else {
+
+                // Handle the case where no weight column was specified
+                // by making the chord weight fixed on both sides.
+                matrix[j][i] = matrix[i][j] = 1;
+            }
+            matrixData[j][i].push(d);
+        });
+
+        matrix.names = names;
+        matrix.data = matrixData;
+
+        return matrix;
+    }
+
+    const matrix = generateMatrix(filterData);
```

Comment:
this seems like an expensive operation let's save it in the state and only change if data is changing. I don't think we need to computed for every re-render.